flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorSubmissionTest.java

```diff
@@ -0,0 +1,1156 @@
+import org.apache.flink.runtime.jobmanager.Tasks;
+import org.apache.flink.runtime.jobmaster.JobMasterId;
+import org.apache.flink.runtime.jobmaster.TestingAbstractInvokables;
+import org.apache.flink.runtime.jobmaster.utils.TestingJobMasterGateway;
+import org.apache.flink.runtime.jobmaster.utils.TestingJobMasterGatewayBuilder;
+import org.apache.flink.runtime.messages.Acknowledge;
+import org.apache.flink.runtime.messages.StackTraceSampleResponse;
+import org.apache.flink.runtime.metrics.groups.UnregisteredMetricGroups;
+import org.apache.flink.runtime.registration.RetryingRegistrationConfiguration;
+import org.apache.flink.runtime.resourcemanager.utils.TestingResourceManagerGateway;
+import org.apache.flink.runtime.rpc.RpcUtils;
+import org.apache.flink.runtime.state.TaskExecutorLocalStateStoresManager;
+import org.apache.flink.runtime.taskexecutor.exceptions.TaskException;
+import org.apache.flink.runtime.taskexecutor.slot.SlotOffer;
+import org.apache.flink.runtime.taskexecutor.slot.TaskSlotTable;
+import org.apache.flink.runtime.testtasks.BlockingNoOpInvokable;
+import org.apache.flink.runtime.testtasks.NoOpInvokable;
+import org.apache.flink.runtime.testutils.StoppableInvokable;
+import org.apache.flink.util.ExceptionUtils;
+import org.apache.flink.util.NetUtils;
+import org.apache.flink.util.SerializedValue;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Tests for submission logic of the {@link TaskExecutor}.
+ */
+public class TaskExecutorSubmissionTest extends TaskExecutorTestBase {
+
+	/**
+	 * Tests that we can submit a task to the TaskManager given that we've allocated a slot there.
+	 */
+	@Test(timeout = 10000L)
+	public void testTaskSubmission() throws Throwable {
+		final JobMasterId jobMasterId = JobMasterId.generate();
+
+		final ExecutionAttemptID eid = new ExecutionAttemptID();
+
+		final TaskDeploymentDescriptor tdd = createTestTaskDeploymentDescriptor("test task", eid, TaskExecutorTest.TestInvokable.class, 1);
+
+		final CompletableFuture<Void> eidRunningFuture = new CompletableFuture<>();
+
+		TestTaskManagerActions taskManagerActions = new TestTaskManagerActions();
+		taskManagerActions.addListener(eid, ExecutionState.RUNNING, eidRunningFuture);
+
+		prepareTaskSubmissionTestEnvironmentBuilder()
+			.setTaskManagerActions(taskManagerActions)
+			.setJobMasterId(jobMasterId)
+			.setConfiguration(this.configuration)
```

Comment:
this is unnecessary since the test isn't mutating the configuration, as is the case for all other tests in this class.

Reply:
There are tests like testRemotePartitionNotFound and testLocalPartitionNotFound need to use a specified configuration, but I can make it use this.configuration as the default.

Reply:
right, but only these 2 tests should actually set a configuration